src/config/tests.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+use std::io::Write;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::sync::{Arc, Condvar, Mutex};
+
+use fvm_shared::address::Address;
+use indoc::formatdoc;
+use ipc_sdk::subnet_id::{SubnetID, ROOTNET_ID};
+use tempfile::NamedTempFile;
+use url::Url;
+
+use crate::config::{Config, ReloadableConfig};
+
+// Arguments for the config's fields
+const SERVER_JSON_RPC_ADDR: &str = "127.0.0.1:3030";
+const ROOT_ID: &str = "/root";
+const CHILD_ID: &str = "/root/f0100";
+const ROOT_AUTH_TOKEN: &str = "ROOT_AUTH_TOKEN";
+const CHILD_AUTH_TOKEN: &str = "CHILD_AUTH_TOKEN";
+const JSONRPC_API_HTTP: &str = "https://example.org/rpc/v0";
+const JSONRPC_API_WS: &str = "ws://example.org/rpc/v0";
+const ACCOUNT_ADDRESS: &str =
+    "f3thgjtvoi65yzdcoifgqh6utjbaod3ukidxrx34heu34d6avx6z7r5766t5jqt42a44ehzcnw3u5ehz47n42a";
+
+#[tokio::test]
+async fn reload_works() {
+    let config_str = config_str();
+
+    let mut file = NamedTempFile::new().unwrap();
+    let path = file
+        .path()
+        .as_os_str()
+        .to_os_string()
+        .into_string()
+        .unwrap();
+
+    file.write_all(config_str.as_bytes()).unwrap();
+
+    let h = Arc::new(ReloadableConfig::new(path.clone()).unwrap());
+    let original_config = h.get_config();
+
+    let pair = Arc::new((Mutex::new(false), Condvar::new()));
```

Comment:
I think it'll help if you add a quick comment describing what you are doing here with the lock and the atomic var (in the end gow you imlpemented the barrier) .